src/Documents.jl

```diff
@@ -134,6 +134,31 @@ immutable EvalNode
     result :: Any
 end
 
+# Navigation
+# ----------------------
+
+"""
+Element in the navigation tree of a document, containing navigation references
+to other page, reference to the [`Page`](@ref) object etc.
+"""
+type NavNode
+    page           :: Nullable{Compat.UTF8String} # unless null, should point to a valid Page in the document
+    title_override :: Nullable{Compat.UTF8String}
+    parent         :: Nullable{NavNode}
+    children       :: Vector{NavNode}
+    prev           :: Nullable{NavNode}
+    next           :: Nullable{NavNode}
+end
+NavNode(page, title_override, parent) = NavNode(page, title_override, parent, [], nothing, nothing)
+
+"""
+Constructs a list of the parents of the `navnode`, ordered such that the `navnode`
+itself is the last item.
+"""
+parents(navnode::NavNode) = isnull(navnode.parent) ? [navnode] :
+    push!(parents(get(navnode.parent)), navnode)
```

Comment:
Does `parents` return all the `NavNode`s all the way up until root or just the single `.parent` of a node?